### PR TITLE
Change plugin file input to be based on file extension

### PR DESCRIPTION
**Pull request by stopalopa**
https://issues.jenkins-ci.org/browse/JENKINS-58821

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,8 +14,7 @@ java -jar plugin-management-cli/target/jenkins-plugin-manager-*.jar --war /file/
 ```
 
 #### CLI Options
-* `--plugin-file` or `-f`: (optional) Path to plugins.txt, which contains a list of plugins to install. If this file does not exist, option will be ignored.
-* `--plugin-yaml` or `-y`: (optional) Path to yaml file containing plugins to install.
+* `--plugin-file` or `-f`: (optional) Path to the plugins.txt or plugins.yaml file, which contains a list of plugins to install. If this file does not exist, an error will be thrown. If the file exists, but does not have a .txt or .yaml/.yml extension, the file will be ignored.
```
Comment (oleg-nenashev) [inline README.md]: Throw error instead?

File: plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java [hunk 2]
```diff
@@ -176,8 +165,17 @@ private String getJenkinsWar() {
...
-        requestedPlugins.addAll(pluginParser.parsePluginYamlFile(getPluginYaml()));
+
+        File pluginFile = getPluginFile();
+        if (pluginFile != null) {
+            String fileExtension = FilenameUtils.getExtension(pluginFile.toString());
+            if (fileExtension.equals("yaml") || fileExtension.equals("yml")) {
+                requestedPlugins.addAll(pluginParser.parsePluginYamlFile(pluginFile));
+            } else if (fileExtension.equals("txt")) {
+                requestedPlugins.addAll(pluginParser.parsePluginTxtFile(pluginFile));
+            }
```
Comment (timja) [inline plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java]: else throw exception?

File: plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java [hunk 3]
```diff
@@ -176,8 +165,17 @@ private String getJenkinsWar() {
         List<Plugin> requestedPlugins = new ArrayList<>();
         PluginParser pluginParser = new PluginParser();
         requestedPlugins.addAll(pluginParser.parsePluginsFromCliOption(plugins));
-        requestedPlugins.addAll(pluginParser.parsePluginTxtFile(getPluginTxt()));
-        requestedPlugins.addAll(pluginParser.parsePluginYamlFile(getPluginYaml()));
+
+        File pluginFile = getPluginFile();
+        if (pluginFile != null) {
+            String fileExtension = FilenameUtils.getExtension(pluginFile.toString());
+            if (fileExtension.equals("yaml") || fileExtension.equals("yml")) {
```
Comment (timja) [inline plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java]: maybe extract to an isYaml function?

**Comment by kwhetstone** [comment]
Is there a ticket for this?

**Comment by stopalopa** [comment]
> Is there a ticket for this?

Yes, I created one: https://issues.jenkins-ci.org/browse/JENKINS-58821

File: plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java [hunk 4]
```diff
@@ -109,31 +106,23 @@ Config setup() {
...
-    private File getPluginYaml() {
-        if (pluginYaml == null) {
-            System.out.println("No .yaml file containing list of plugins to be downloaded entered.");
-        } else {
-            System.out.println("Yaml file containing list of plugins to be downloaded: " + pluginYaml);
+            if (Files.exists(pluginFile.toPath())) {
+                System.out.println("File containing list of plugins to be downloaded: " + pluginFile);
+            }
+            else {
+                throw new PluginInputException("File containing list of plugins does not exist");
```
Comment (timja) [inline plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java]: Include the file path please,
```suggestion
                throw new PluginInputException("File containing list of plugins does not exist " + pluginFile.toPath());
```

File: plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java [hunk 5]
```diff
@@ -109,31 +106,23 @@ Config setup() {
...
-     * @return plugin yaml file passed in through CLI, or null if user did not pass in a yaml file
-     */
-    private File getPluginYaml() {
-        if (pluginYaml == null) {
-            System.out.println("No .yaml file containing list of plugins to be downloaded entered.");
-        } else {
-            System.out.println("Yaml file containing list of plugins to be downloaded: " + pluginYaml);
+            if (Files.exists(pluginFile.toPath())) {
+                System.out.println("File containing list of plugins to be downloaded: " + pluginFile);
+            }
```
Comment (timja) [inline plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java]: else should be on the same line
```suggestion
            } else {
```

File: plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java [hunk 6]
```diff
@@ -109,31 +106,23 @@ Config setup() {
...
+            else {
```
Comment (timja) [inline plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java]:
```suggestion
```

**Pull request merged**